ShopperInsights/src/main/java/com/braintreepayments/api/shopperinsights/v2/ShopperInsightsClientV2.kt

```diff
@@ -31,9 +38,62 @@ class ShopperInsightsClientV2 internal constructor(
         context: Context,
         authorization: String
     ) : this(
-        BraintreeClient(context, authorization),
-        lazyAnalyticsClient = AnalyticsClient.lazyInstance
+        BraintreeClient(context, authorization)
     )
 
     private val analyticsClient: AnalyticsClient by lazyAnalyticsClient
+
+    /**
+     * Call this method when the PayPal, Venmo or Other button has been successfully displayed to the buyer.
+     * This method sends analytics to help improve the Shopper Insights feature experience.
+     * @param buttonType Type of button presented - PayPal, Venmo, or Other.
+     * @param presentmentDetails Detailed information, including button order, experiment type, and page type about the
+     * payment button that is sent to analytics to help improve the Shopper Insights feature experience.
+     * @param sessionId The shopper session ID
+     */
+    fun sendPresentedEvent(
```

Comment:
Do we want to consider having these all defined in 1 place that can be shared? Thinking of an instance where we have 3-4 clients all with the same methods and want to add something to all of them, we would need to manually make the change in 3-4 places. Not sure if that's feasible or not though, so take it or leave it!

Reply:
I felt that there wasn't much going on in the function - mapping `PresentmentDetails` to an `AnalyticsEventParams` type and sending the event. I think if the mapping became more complicated we could create a use case that could be used in all Clients.